src/utils/themeToCSSVariables.ts

```diff
@@ -0,0 +1,17 @@
+import type { CSSVarName, CSSVariables, ThemeColors, ThemeName } from '$types';
+import { themes } from '$data/themes';
+
+export const themeToCSSVariables = (themeName: ThemeName): string => {
+	const themeColors = themes[themeName];
+
+	// keys to css vars
+	const colorKeys = Object.keys(themeColors) as [keyof ThemeColors];
+	const cssVars = colorKeys.reduce((prev, color) => {
+		prev[`--main-${color}-color`] = themeColors[color];
+		return prev;
+	}, {} as CSSVariables);
+
+	// construct CSS-injectable string
+	const cssVarsKeys = Object.keys(cssVars) as CSSVarName[];
+	return cssVarsKeys.reduce((prev, curr) => prev.concat(`${curr}:${cssVars[curr]};\n`), '');
+};
```

Comment:
предлагаю выделять блоки в отдельные методы, вместо комментов

```suggestion
import type { CSSVarName, CSSVariables, ThemeColors, ThemeName } from '$types';
import { themes } from '$data/themes';

export const themeToCSSVariables = (themeName: ThemeName): string => {
	const themeColors = themes[themeName];

	const cssVars = getCSSVarsFromThemeColors(themeColors)
	return constructCSSVarsString(cssVars);
};

const getCSSVarsFromThemeColors = (colors: ThemeColors): CSSVariables => {
	const colorKeys = Object.keys(colors) as [keyof ThemeColors];
	const cssVars = colorKeys.reduce((prev, color) => {
		prev[`--main-${color}-color`] = colors[color];
		return prev;
	}, {} as CSSVariables);
	return cssVars;
};

const constructCSSVarsString = (cssVars: CSSVariables): string => {
	const cssVarsKeys = Object.keys(cssVars) as CSSVarName[];
	return cssVarsKeys.reduce((prev, curr) => prev.concat(`${curr}:${cssVars[curr]};\n`), '');
};
```